services/swap/price.ts

```diff
@@ -48,29 +49,48 @@ export const getToken2ForToken1Price = async ({
 
 export interface GetTokenForTokenPriceInput {
   tokenAmount: number
-  swapAddress: string
-  outputSwapAddress: string
+  tokenA: TokenInfo
+  tokenB: TokenInfo
+  swap: PoolEntityType
+  outputSwap: PoolEntityType
   client: CosmWasmClient
 }
 
 export const getTokenForTokenPrice = async ({
   tokenAmount,
-  swapAddress,
-  outputSwapAddress,
+  tokenA,
+  tokenB,
+  swap,
+  outputSwap,
   client,
 }: GetTokenForTokenPriceInput) => {
   try {
-    const nativePrice = await getToken2ForToken1Price({
-      tokenAmount,
-      swapAddress,
-      client,
-    })
+    const intermediatePrice =
+      tokenA.symbol === swap.pool_assets[0].symbol
+        ? await getToken1ForToken2Price({
+            nativeAmount: tokenAmount,
+            swapAddress: swap.swap_address,
+            client,
+          })
+        : await getToken2ForToken1Price({
+            tokenAmount,
+            swapAddress: swap.swap_address,
+            client,
+          })
 
-    return getToken1ForToken2Price({
-      nativeAmount: nativePrice,
-      swapAddress: outputSwapAddress,
-      client: client,
-    })
+    let result =
```

Comment:
Nice! Looks cleaner this way. Can also just return this entire condition out the function and skip this assignment

Reply:
good catch, was doing the assignment becuase i was logging the variable at one point